### ISSUE TITLE
Incorrect JSON produced for accessor `constructor`

### ISSUE THREAD

**Issue by arnog** [TypeStrong/typedoc#1280]
When an accessor named "constructor" is used in a class, the JSON output is unexpected.

##  Expected Behavior
A separate entry for the class constructor (kind: 512) and for the accessor named "constructor" (kind : 262144)

## Actual Behavior
A single constructor entry (kind 512) with both a `signatures` and `getSignature` properties.

## Steps to reproduce the bug
Run typedoc on this input:
```typescript
declare class C {
  constructor();
  get constructor(): boolean;
}
```
the resulting JSON is:
```json
{
	"id": 0,
	"kind": 0,
	"children": [
		{
			"id": 1,
			"kind": 1,
			"kindString": "Module",
			"flags": {
				"isExported": true
			},
			"children": [
				{
					"id": 2,
					"name": "C",
					"kind": 128,
					"kindString": "Class",
					"flags": {
						"isExported": true
					},
					"children": [
						{
							"id": 3,
							"name": "constructor",
							"kind": 512,
							"kindString": "Constructor",
							"flags": {
								"isExported": true
							},
							"signatures": [
								{
									"id": 4,
									"name": "new C",
									"kind": 16384,
									"kindString": "Constructor signature",
									"flags": {
										"isExported": true
									},
									"type": {
										"type": "reference",
										"id": 2,
										"name": "C"
									}
								}
							],
							"getSignature": [
								{
									"id": 5,
									"name": "__get",
									"kind": 524288,
									"kindString": "Get signature",
									"flags": {
										"isExported": true
									},
									"type": {
										"type": "intrinsic",
										"name": "boolean"
									}
								}
							]
						}
					],
					"groups": [
						{
							"title": "Constructors",
							"kind": 512,
							"children": [
								3
							]
						}
					],
				}
			],
			"groups": [
				{
					"title": "Classes",
					"kind": 128,
					"children": [
						2
					]
				}
			]
		}
	],
	"groups": [
		{
			"title": "Modules",
			"kind": 1,
			"children": [
				1
			]
		}
	]
}
```

**Comment by Gerrit0** [comment]
Hmm, this is kind of a tricky one. I believe it's another case of TypeDoc trying to do declaration merging, and doing it.. badly. #1244. PR welcome if you'd like to give fixing it a shot, otherwise I'll add this to my test cases to resolve before library mode is done.

**Comment by Gerrit0** [comment]
This is fixed in 0.20 (currently `typedoc@beta`) - we finally stopped doing bad things by doing declaration merging ourselves!

![image](https://user-images.githubusercontent.com/19329837/100492022-c115af80-30e5-11eb-8f18-d8e606d7658b.png)
